asmcomp/cmm_helpers.ml

```diff
@@ -151,6 +151,12 @@ let rec add_const c n dbg =
 let incr_int c dbg = add_const c 1 dbg
 let decr_int c dbg = add_const c (-1) dbg
 
+let add_int_addr c1 c2 dbg =
+  match c1, c2 with
+  | Cconst_int (0, _), c | c, Cconst_int (0, _) -> c
+  | Cconst_natint (0n, _), c | c, Cconst_natint (0n, _) -> c
+  | _, _ -> Cop (Cadda, [c1; c2], dbg)
+
```

Comment:
```suggestion
let offset_addr c1 c2 dbg =
  match c1, c2 with
  | c, Cconst_int (0, _) -> c
  | c, Cconst_natint (0n, _) -> c
  | Cop(Cadda, [c; Cconst_int(n1, _)], _), _ ->
      Cop(Cadda, [c; add_const c2 n1 dbg], dbg)
  | _, _ -> Cop (Cadda, [c1; c2], dbg)
```
Concrete proposal: 1- rename, 2- break symmetry, 3- add optimization for `(addr + cst) + idx` case, reassociated into `addr + optimized(cst + idx)`.

Reply:
Thanks, will apply a change of this form.